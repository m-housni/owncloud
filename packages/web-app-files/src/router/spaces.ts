import { Location, RouteConfig } from 'vue-router'
import { RouteComponents } from './router'
import { createLocation, isLocationActiveDirector, $gettext } from './utils'

type spaceTypes = 'files-spaces-projects' | 'files-spaces-generic'

export const createLocationSpaces = (name: spaceTypes, location = {}): Location =>
  createLocation(name, location)

export const locationSpacesProjects = createLocationSpaces('files-spaces-projects')
export const locationSpacesGeneric = createLocationSpaces('files-spaces-generic')

// FIXME: `isLocationSpacesActive('files-spaces-generic') returns true for 'files-spaces-projects' as well
// TODO: if that's fixed, adjust the `loaderSpaceGeneric#isActive` and `loaderShare#isActive`
export const isLocationSpacesActive = isLocationActiveDirector<spaceTypes>(
  locationSpacesProjects,
  locationSpacesGeneric
)

export const buildRoutes = (components: RouteComponents): RouteConfig[] => [
  {
    path: '/spaces',
    component: components.App,
    children: [
      {
        path: 'projects',
        name: locationSpacesProjects.name,
        component: components.Spaces.Projects,
        meta: {
          authContext: 'user',
          title: $gettext('Spaces')
        }
      },
      {
        path: ':driveAliasAndItem*',
        name: locationSpacesGeneric.name,
        component: components.Spaces.DriveResolver,
        meta: {
          authContext: 'user',
          patchCleanPath: true
        }
      }
    ]
  }
]
