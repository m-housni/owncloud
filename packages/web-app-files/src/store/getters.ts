import { isProjectSpaceResource } from 'web-client/src/helpers'
import { ShareTypes } from 'web-client/src/helpers/share'

export default {
  selectedFiles: (state, getters) => {
    return getters.filesAll.filter((f) => state.selectedIds.includes(f.id))
  },
  files: (state) => {
    return state.files
  },
  filesAll: (state) => state.filesSearched || state.files,
  currentFolder: (state) => {
    return state.currentFolder
  },
  clipboardResources: (state) => {
    return state.clipboardResources
  },
  clipboardAction: (state) => {
    return state.clipboardAction
  },
  activeFiles: (state, getters) => {
    let files = [].concat(getters.filesAll)

    if (!state.areHiddenFilesShown) {
      files = files.filter((file) => !file.name.startsWith('.'))
    }

    return files
  },
  totalFilesSize: (state, getters) => {
    return getters.filesAll.map((file) => parseInt(file.size)).reduce((x, y) => x + y, 0)
  },
  totalFilesCount: (state, getters) => {
    const fileCount = getters.filesAll.filter((file) => file.type === 'file').length
    const folderCount = getters.filesAll.filter((file) => file.type === 'folder').length
    const spaceCount = getters.filesAll.filter((file) => isProjectSpaceResource(file)).length
    return {
      files: fileCount,
      folders: folderCount,
      spaces: spaceCount
    }
  },
  currentFileOutgoingCollaborators: (state) => {
    return state.currentFileOutgoingShares.filter((share) => {
      return ShareTypes.containsAnyValue(ShareTypes.authenticated, [share.shareType])
    })
  },
  currentFileOutgoingLinks: (state) => {
    return state.currentFileOutgoingShares.filter((share) => {
      return ShareTypes.containsAnyValue(ShareTypes.unauthenticated, [share.shareType])
    })
  },
  sharesTree: (state) => state.sharesTree,
  sharesTreeLoading: (state) => state.sharesTreeLoading,
  quota: (state) => {
    return state.quota
  },
  highlightedFile: (state, getters) => {
    if (getters.selectedFiles.length > 0) {
      return getters.selectedFiles[0]
    }
    return state.currentFolder
  },
  versions: (state) => {
    return state.versions
  }
}
