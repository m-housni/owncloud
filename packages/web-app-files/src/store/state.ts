export default {
  currentFolder: null,
  files: [],
  filesSearched: null,
  selectedIds: [],
  latestSelectedId: null,
  clipboardSpace: null,
  clipboardResources: [],
  clipboardAction: null,
  versions: [],

  /**
   * Outgoing shares and links from currently highlighted element
   */
  currentFileOutgoingShares: [],

  /**
   * Incoming shares from currently highlighted element
   */
  incomingShares: [],

  /**
   * Shares from parent folders
   **/
  sharesTree: {},
  sharesTreeError: null,
  sharesTreeLoading: false,

  /**
   * View settings
   */
  areHiddenFilesShown: true,
  areFileExtensionsShown: true
}
