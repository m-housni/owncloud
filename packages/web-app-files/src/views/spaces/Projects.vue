<template>
  <div class="oc-flex">
    <files-view-wrapper>
      <app-bar
        :breadcrumbs="breadcrumbs"
        :has-view-options="false"
        :has-sidebar-toggle="true"
        :show-actions-on-selection="true"
        :side-bar-open="sideBarOpen"
      >
        <template #actions>
          <create-space v-if="hasCreatePermission" />
        </template>
      </app-bar>
      <app-loading-spinner v-if="areResourcesLoading" />
      <template v-else>
        <no-content-message
          v-if="!spaces.length"
          id="files-spaces-empty"
          class="files-empty"
          icon="layout-grid"
        >
          <template #message>
            <span v-translate>You don't have access to any spaces</span>
          </template>
        </no-content-message>
        <div v-else class="spaces-list oc-px-m oc-mt-l">
          <oc-list class="oc-flex">
            <li
              v-for="space in spaces"
              :key="space.getDomSelector()"
              class="spaces-list-item oc-mb-m oc-mr-m"
            >
              <div
                class="spaces-list-card oc-card oc-card-default oc-rounded"
                :data-item-id="space.id"
                :class="getSpaceCardAdditionalClass(space)"
                @contextmenu="(event) => showSpaceContextDrop(event, space)"
              >
                <div class="oc-card-media-top oc-border-b">
                  <component
                    :is="space.disabled ? 'oc-button' : 'router-link'"
                    v-bind="getSpaceLinkProps(space)"
                    v-on="getSpaceLinkListeners(space)"
                  >
                    <oc-tag
                      v-if="space.disabled"
                      class="oc-position-absolute space-disabled-indicator"
                      type="span"
                    >
                      <span v-translate>Disabled</span>
                    </oc-tag>
                    <img
                      v-if="imageContentObject[space.id]"
                      class="space-image oc-rounded-top"
                      :src="imageContentObject[space.id]['data']"
                      alt=""
                    />
                    <oc-icon
                      v-else
                      name="layout-grid"
                      size="xxlarge"
                      class="space-default-image oc-px-m oc-py-m"
                    />
                  </component>
                </div>
                <div class="oc-card-body oc-p-s">
                  <div class="oc-flex oc-flex-between oc-flex-middle">
                    <div class="oc-flex oc-flex-middle oc-text-truncate">
                      <oc-icon class="oc-mr-s" name="layout-grid" />
                      <component
                        :is="space.disabled ? 'oc-button' : 'router-link'"
                        v-bind="getSpaceLinkProps(space)"
                        class="space-name oc-text-left oc-text-truncate"
                        v-on="getSpaceLinkListeners(space)"
                      >
                        <span v-text="space.name"> </span>
                      </component>
                    </div>
                    <div class="oc-flex oc-flex-middle">
                      <div>
                        <oc-button
                          v-oc-tooltip="$gettext('Show members')"
                          :aria-label="$gettext('Show members')"
                          appearance="raw"
                          @click="openSidebarSharePanel(space)"
                        >
                          <oc-icon name="group" fill-type="line" />
                        </oc-button>
                      </div>
                      <div>
                        <oc-button
                          :id="`space-context-btn-${space.getDomSelector()}`"
                          :ref="`spaceContextBtn-${space.getDomSelector()}`"
                          v-oc-tooltip="$gettext('Show context menu')"
                          :aria-label="$gettext('Show context menu')"
                          appearance="raw"
                          @click="(event) => showSpaceContextDrop(event, space)"
                        >
                          <oc-icon name="more-2" />
                        </oc-button>
                        <oc-drop
                          :ref="`spaceContextDrop-${space.getDomSelector()}`"
                          :drop-id="`space-context-drop-${space.getDomSelector()}`"
                          mode="manual"
                          close-on-click
                          :options="{ delayHide: 0 }"
                          padding-size="small"
                          position="bottom-end"
                        >
                          <space-context-actions :items="[space]" :space="space" />
                        </oc-drop>
                      </div>
                    </div>
                  </div>
                  <p
                    v-if="space.description"
                    class="oc-text-left oc-ml-xs oc-mt-xs oc-mb-rm oc-text-truncate"
                  >
                    <small v-text="space.description"></small>
                  </p>
                </div>
              </div>
            </li>
          </oc-list>
        </div>
      </template>
    </files-view-wrapper>
    <side-bar :open="sideBarOpen" :active-panel="sideBarActivePanel" :space="highlightedFile" />
  </div>
</template>

<script lang="ts">
import AppBar from '../../components/AppBar/AppBar.vue'
import CreateSpace from '../../components/AppBar/CreateSpace.vue'
import NoContentMessage from 'web-pkg/src/components/NoContentMessage.vue'
import AppLoadingSpinner from 'web-pkg/src/components/AppLoadingSpinner.vue'
import { computed, defineComponent, unref } from 'vue'
import { useAccessToken, useStore, useGraphClient } from 'web-pkg/src/composables'
import { useTask } from 'vue-concurrency'
import { mapMutations, mapActions, mapGetters } from 'vuex'
import { loadPreview } from 'web-pkg/src/helpers/preview'
import { ImageDimension } from '../../constants'
import SpaceContextActions from '../../components/Spaces/SpaceContextActions.vue'
import { configurationManager } from 'web-pkg/src/configuration'
import { isProjectSpaceResource, SpaceResource } from 'web-client/src/helpers'
import SideBar from '../../components/SideBar/SideBar.vue'
import FilesViewWrapper from '../../components/FilesViewWrapper.vue'
import { eventBus } from 'web-pkg/src/services/eventBus'
import { SideBarEventTopics, useSideBar } from '../../composables/sideBar'
import { WebDAV } from 'web-client/src/webdav'
import { createLocationSpaces } from '../../router'
import { createFileRouteOptions } from 'web-pkg/src/helpers/router'
import { useScrollTo } from 'web-app-files/src/composables/scrollTo'

export default defineComponent({
  components: {
    FilesViewWrapper,
    SideBar,
    AppBar,
    AppLoadingSpinner,
    CreateSpace,
    NoContentMessage,
    SpaceContextActions
  },
  setup() {
    const store = useStore()
    const spaces = computed(
      () =>
        store.getters['runtime/spaces/spaces']
          .filter((s) => isProjectSpaceResource(s))
          .sort((a, b) => a.name.localeCompare(b.name)) || []
    )
    const accessToken = useAccessToken({ store })
    const { graphClient } = useGraphClient()

    const loadResourcesTask = useTask(function* () {
      store.commit('Files/CLEAR_FILES_SEARCHED')
      store.commit('Files/CLEAR_CURRENT_FILES_LIST')
      yield store.dispatch('runtime/spaces/reloadProjectSpaces', {
        graphClient: unref(graphClient)
      })
      store.commit('Files/LOAD_FILES', { currentFolder: null, files: unref(spaces) })
    })

    const areResourcesLoading = computed(() => {
      return loadResourcesTask.isRunning || !loadResourcesTask.last
    })

    return {
      ...useSideBar(),
      ...useScrollTo(),
      spaces,
      graphClient,
      loadResourcesTask,
      areResourcesLoading,
      accessToken
    }
  },
  data: function () {
    return {
      imageContentObject: {}
    }
  },
  computed: {
    ...mapGetters(['user']),
    ...mapGetters('Files', ['highlightedFile']),
    breadcrumbs() {
      return [
        {
          text: this.$gettext('Spaces'),
          onClick: () => this.loadResourcesTask.perform()
        }
      ]
    },
    hasCreatePermission() {
      return this.$permissionManager.hasSpaceManagement()
    }
  },
  watch: {
    spaces: {
      handler: async function (val) {
        if (!val) {
          return
        }

        for (const space of this.spaces as SpaceResource[]) {
          if (!space.spaceImageData) {
            continue
          }

          if (this.imageContentObject[space.id]?.fileId === space.spaceImageData?.id) {
            continue
          }

          const decodedUri = decodeURI(space.spaceImageData.webDavUrl)
          const webDavPathComponents = decodedUri.split('/')
          const idComponent = webDavPathComponents.find((c) => c.startsWith(`${space.id}`))
          if (!idComponent) {
            return
          }
          const path = webDavPathComponents
            .slice(webDavPathComponents.indexOf(idComponent) + 1)
            .join('/')

          const resource = await (this.$clientService.webdav as WebDAV).getFileInfo(space, { path })
          loadPreview({
            resource,
            isPublic: false,
            dimensions: ImageDimension.Preview,
            server: configurationManager.serverUrl,
            userId: this.user.id,
            token: this.accessToken
          }).then((imageBlob) => {
            this.$set(this.imageContentObject, space.id, {
              fileId: space.spaceImageData.id,
              data: imageBlob
            })
          })
        }
      },
      deep: true,
      immediate: true
    }
  },
  async created() {
    await this.loadResourcesTask.perform()
    this.scrollToResourceFromRoute(this.spaces)
  },
  methods: {
    ...mapActions(['showMessage']),
    ...mapMutations('Files', ['SET_CURRENT_FOLDER', 'SET_FILE_SELECTION']),

    showSpaceContextDrop(event, space) {
      event.preventDefault()
      const drop = this.$refs[`spaceContextDrop-${space.getDomSelector()}`]?.[0]
      const contextMenuButtonPos =
        this.$refs[`spaceContextBtn-${space.getDomSelector()}`]?.[0].$el?.getBoundingClientRect()

      if (!drop || !contextMenuButtonPos) {
        return
      }

      drop.tippy.setProps({
        getReferenceClientRect: () => ({
          width: 0,
          height: 0,
          top: event.clientY,
          bottom: event.clientY,
          left: event.type === 'contextmenu' ? event.clientX : contextMenuButtonPos.x,
          right: event.type === 'contextmenu' ? event.clientX : contextMenuButtonPos.x
        })
      })
      drop.show()
    },
    getSpaceProjectRoute(space: SpaceResource) {
      return space.disabled
        ? '#'
        : createLocationSpaces(
            'files-spaces-generic',
            createFileRouteOptions(space, { path: '', fileId: space.fileId })
          )
    },

    getSpaceCardAdditionalClass(space) {
      if (space.disabled) {
        return 'state-trashed'
      }
      return ''
    },

    openSidebarSharePanel(space: SpaceResource) {
      this.SET_FILE_SELECTION([space])
      eventBus.publish(SideBarEventTopics.openWithPanel, 'space-share')
    },

    getSpaceLinkProps(space: SpaceResource) {
      if (space.disabled) {
        return {
          appearance: 'raw'
        }
      }
      return { to: this.getSpaceProjectRoute(space) }
    },

    getSpaceLinkListeners(space) {
      if (!space.disabled) {
        return {}
      }

      return {
        click: () => {
          this.showMessage({
            title: this.$gettext('Disabled spaces cannot be entered'),
            status: 'warning'
          })
        }
      }
    }
  }
})
</script>

<style lang="scss">
#files-spaces-empty {
  height: 75vh;
}

.spaces-list {
  ul {
    flex-wrap: wrap;
  }

  &-item {
    width: 252px;
  }

  &-card {
    box-shadow: none !important;
    background-color: var(--oc-color-background-highlight) !important;
    height: 100%;
  }

  &-card.state-trashed {
    .space-image,
    .space-default-image > svg {
      filter: grayscale(100%);
      opacity: 80%;
    }
  }

  .oc-card-media-top {
    aspect-ratio: 16/9;
  }

  .oc-card-media-top a {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 100%;
  }

  .oc-card-media-top button {
    width: 100%;
    height: 100%;
  }

  .oc-tag {
    color: var(--oc-color-text-default);
  }

  .space-image {
    width: 100%;
    aspect-ratio: 16/9;
    object-fit: cover;
  }

  .space-name {
    overflow: hidden;
    color: var(--oc-color-text-default);
  }

  .space-disabled-indicator {
    z-index: 999;
  }

  .spaces-list-projects {
    aspect-ratio: 16/9;
  }
}
</style>
