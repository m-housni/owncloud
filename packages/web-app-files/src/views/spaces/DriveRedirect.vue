<template>
  <div class="oc-flex oc-width-1-1">
    <app-loading-spinner />
  </div>
</template>

<script lang="ts">
import { computed, defineComponent, unref } from 'vue'
import { useRoute, useRouter, useStore } from 'web-pkg/src/composables'
import AppLoadingSpinner from 'web-pkg/src/components/AppLoadingSpinner.vue'
import { urlJoin } from 'web-client/src/utils'
import { createFileRouteOptions } from 'web-pkg/src/helpers/router'

// 'personal/home' is used as personal drive alias from static contexts
// (i.e. places where we can't load the actual personal space)
const fakePersonalDriveAlias = 'personal/home'

export default defineComponent({
  name: 'DriveRedirect',
  components: {
    AppLoadingSpinner
  },
  props: {
    driveAliasAndItem: {
      type: String,
      required: false,
      default: ''
    },
    appendHomeFolder: {
      type: Boolean,
      required: false,
      default: false
    }
  },
  setup(props) {
    const store = useStore()
    const router = useRouter()
    const route = useRoute()

    const personalSpace = computed(() => {
      return store.getters['runtime/spaces/spaces'].find((space) => space.driveType === 'personal')
    })
    const itemPath = computed(() => {
      if (!props.appendHomeFolder) {
        return ''
      }
      const item = props.driveAliasAndItem.startsWith(fakePersonalDriveAlias)
        ? urlJoin(props.driveAliasAndItem.slice(fakePersonalDriveAlias.length))
        : '/'
      if (item !== '/') {
        return item
      }
      return store.getters.homeFolder
    })

    const { params, query } = createFileRouteOptions(unref(personalSpace), {
      path: unref(itemPath)
    })

    return (
      router
        .replace({
          ...unref(route),
          params: {
            ...unref(route).params,
            ...params
          },
          query
        })
        // avoid NavigationDuplicated error in console
        .catch(() => {})
    )
  }
})
</script>
