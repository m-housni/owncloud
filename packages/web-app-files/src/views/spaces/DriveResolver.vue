<template>
  <div class="oc-flex oc-width-1-1">
    <drive-redirect
      v-if="!space"
      :drive-alias-and-item="driveAliasAndItem"
      :append-home-folder="isSpaceRoute"
    />
    <generic-trash v-else-if="isTrashRoute" :space="space" :item-id="itemId" />
    <generic-space v-else :space="space" :item="item" :item-id="itemId" />
  </div>
</template>

<script lang="ts">
import DriveRedirect from './DriveRedirect.vue'
import GenericSpace from './GenericSpace.vue'
import GenericTrash from './GenericTrash.vue'

import { defineComponent, onMounted, unref } from 'vue'
import {
  useClientService,
  useDriveResolver,
  useRouteParam,
  useRouter,
  useStore
} from 'web-pkg/src/composables'
import { useActiveLocation } from '../../composables'
import { isLocationSpacesActive, isLocationTrashActive } from '../../router'
import { isPublicSpaceResource, PublicSpaceResource, SpaceResource } from 'web-client/src/helpers'
import { locationPublicUpload } from 'web-app-files/src/router/public'
import { linkRoleUploaderFolder } from 'web-client/src/helpers/share'

export default defineComponent({
  components: {
    DriveRedirect,
    GenericSpace,
    GenericTrash
  },
  setup() {
    const store = useStore()
    const clientService = useClientService()
    const router = useRouter()
    const driveAliasAndItem = useRouteParam('driveAliasAndItem')
    const isSpaceRoute = useActiveLocation(isLocationSpacesActive, 'files-spaces-generic')
    const isTrashRoute = useActiveLocation(isLocationTrashActive, 'files-trash-generic')
    const resolvedDrive = useDriveResolver({ store, driveAliasAndItem })

    const getSpaceResource = async (): Promise<SpaceResource> => {
      const space = unref(resolvedDrive.space)
      try {
        return (await clientService.webdav.getFileInfo(space)) as SpaceResource
      } catch (e) {
        console.error(e)
        return space
      }
    }

    onMounted(async () => {
      const space = unref(resolvedDrive.space)
      if (space && isPublicSpaceResource(space)) {
        let publicSpace = (await getSpaceResource()) as PublicSpaceResource

        if (linkRoleUploaderFolder.bitmask(false) === publicSpace.publicLinkPermission) {
          router.push({
            name: locationPublicUpload.name,
            params: { token: space.id.toString() }
          })
        }
      }
    })

    return {
      ...resolvedDrive,
      driveAliasAndItem,
      isSpaceRoute,
      isTrashRoute
    }
  }
})
</script>
