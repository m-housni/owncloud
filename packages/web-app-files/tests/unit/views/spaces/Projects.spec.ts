import { mount } from '@vue/test-utils'
import Projects from '../../../../src/views/spaces/Projects.vue'
import { defaultStoreMockOptions } from 'web-test-helpers/src/mocks/store/defaultStoreMockOptions'
import { defaultComponentMocks } from 'web-test-helpers/src/mocks/defaultComponentMocks'
import { createStore } from 'vuex-extensions'
import { defaultLocalVue } from 'web-test-helpers/src/localVue/defaultLocalVue'
import Vuex from 'vuex'
import { spaces } from '../../../__fixtures__/spaces'
import { defaultStubs } from 'web-test-helpers/src/mocks/defaultStubs'

describe('Projects view', () => {
  it('appBar always present', () => {
    const { wrapper } = getMountedWrapper()
    expect(wrapper.find('app-bar-stub').exists()).toBeTruthy()
  })
  it('sideBar always present', () => {
    const { wrapper } = getMountedWrapper()
    expect(wrapper.find('side-bar-stub').exists()).toBeTruthy()
  })
  describe('different files view states', () => {
    it('shows the loading spinner during loading', () => {
      const { wrapper } = getMountedWrapper()
      expect(wrapper.find('oc-spinner-stub').exists()).toBeTruthy()
    })
    it('shows the no-content-message after loading', async () => {
      const { wrapper } = getMountedWrapper()
      await wrapper.vm.loadResourcesTask.last
      expect(wrapper.find('oc-spinner-stub').exists()).toBeFalsy()
      expect(wrapper.find('.no-content-message').exists()).toBeTruthy()
    })
    it('lists all available project spaces', async () => {
      const { wrapper } = getMountedWrapper({ spaces: [spaces[0], spaces[1], spaces[2]] })
      await wrapper.vm.loadResourcesTask.last
      expect(wrapper).toMatchSnapshot()
      expect(wrapper.find('.no-content-message').exists()).toBeFalsy()
      expect(wrapper.find('.spaces-list').exists()).toBeTruthy()
      expect(wrapper.findAll('.spaces-list-item').length).toEqual(3)
    })
  })
})

function getMountedWrapper({ mocks = {}, spaces = [] } = {}) {
  const defaultMocks = {
    ...defaultComponentMocks({
      currentRoute: { name: 'files-spaces-projects' }
    }),
    ...(mocks && mocks)
  }
  const storeOptions = { ...defaultStoreMockOptions }
  storeOptions.modules.runtime.modules.spaces.getters.spaces = jest.fn(() => spaces)
  const localVue = defaultLocalVue()
  const store = createStore(Vuex.Store, storeOptions)
  return {
    mocks: defaultMocks,
    storeOptions,
    wrapper: mount(Projects, {
      localVue,
      mocks: defaultMocks,
      store,
      stubs: defaultStubs
    })
  }
}
