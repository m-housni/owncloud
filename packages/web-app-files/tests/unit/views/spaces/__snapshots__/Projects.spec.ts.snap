// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects view different files view states lists all available project spaces 1`] = `
<div class="oc-flex">
  <div class="files-view-wrapper oc-width-expand">
    <div id="files-view">
      <app-bar-stub breadcrumbs="[object Object]" breadcrumbscontextactionsitems="" hassidebartoggle="true" showactionsonselection="true"></app-bar-stub>
      <div class="spaces-list oc-px-m oc-mt-l">
        <ul class="oc-list oc-my-rm oc-mx-rm oc-flex">
          <li class="spaces-list-item oc-mb-m oc-mr-m">
            <div data-item-id="2" class="spaces-list-card oc-card oc-card-default oc-rounded">
              <div class="oc-card-media-top oc-border-b">
                <router-link-stub tag="a" to="[object Object]">
                  <!----> <span class="space-default-image oc-px-m oc-py-m oc-icon oc-icon-xxl oc-icon-passive"><!----></span>
                </router-link-stub>
              </div>
              <div class="oc-card-body oc-p-s">
                <div class="oc-flex oc-flex-between oc-flex-middle">
                  <div class="oc-flex oc-flex-middle oc-text-truncate"><span class="oc-mr-s oc-icon oc-icon-m oc-icon-passive"><!----></span>
                    <router-link-stub tag="a" to="[object Object]" class="space-name oc-text-left oc-text-truncate"><span>Another space</span></router-link-stub>
                  </div>
                  <div class="oc-flex oc-flex-middle">
                    <div><button aria-label="Show members" type="button" class="oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw"><span class="oc-icon oc-icon-m oc-icon-passive"><!----></span></button></div>
                    <div><button aria-label="Show context menu" type="button" class="oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw" id="space-context-btn-undefined" aria-expanded="false"><span class="oc-icon oc-icon-m oc-icon-passive"><!----></span></button> </div>
                  </div>
                </div>
                <!---->
              </div>
            </div>
          </li>
          <li class="spaces-list-item oc-mb-m oc-mr-m">
            <div data-item-id="3" class="spaces-list-card oc-card oc-card-default oc-rounded state-trashed">
              <div class="oc-card-media-top oc-border-b"><button type="button" class="oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw"><span class="oc-position-absolute space-disabled-indicator oc-tag oc-tag-m oc-tag-span"><span>Disabled</span></span> <span class="space-default-image oc-px-m oc-py-m oc-icon oc-icon-xxl oc-icon-passive"><!----></span></button></div>
              <div class="oc-card-body oc-p-s">
                <div class="oc-flex oc-flex-between oc-flex-middle">
                  <div class="oc-flex oc-flex-middle oc-text-truncate"><span class="oc-mr-s oc-icon oc-icon-m oc-icon-passive"><!----></span> <button type="button" class="space-name oc-text-left oc-text-truncate oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw"><span>Disabled space</span></button></div>
                  <div class="oc-flex oc-flex-middle">
                    <div><button aria-label="Show members" type="button" class="oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw"><span class="oc-icon oc-icon-m oc-icon-passive"><!----></span></button></div>
                    <div><button aria-label="Show context menu" type="button" class="oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw" id="space-context-btn-undefined" aria-expanded="false"><span class="oc-icon oc-icon-m oc-icon-passive"><!----></span></button> </div>
                  </div>
                </div>
                <!---->
              </div>
            </div>
          </li>
          <li class="spaces-list-item oc-mb-m oc-mr-m">
            <div data-item-id="1" class="spaces-list-card oc-card oc-card-default oc-rounded">
              <div class="oc-card-media-top oc-border-b">
                <router-link-stub tag="a" to="[object Object]">
                  <!----> <span class="space-default-image oc-px-m oc-py-m oc-icon oc-icon-xxl oc-icon-passive"><!----></span>
                </router-link-stub>
              </div>
              <div class="oc-card-body oc-p-s">
                <div class="oc-flex oc-flex-between oc-flex-middle">
                  <div class="oc-flex oc-flex-middle oc-text-truncate"><span class="oc-mr-s oc-icon oc-icon-m oc-icon-passive"><!----></span>
                    <router-link-stub tag="a" to="[object Object]" class="space-name oc-text-left oc-text-truncate"><span>Some space</span></router-link-stub>
                  </div>
                  <div class="oc-flex oc-flex-middle">
                    <div><button aria-label="Show members" type="button" class="oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw"><span class="oc-icon oc-icon-m oc-icon-passive"><!----></span></button></div>
                    <div><button aria-label="Show context menu" type="button" class="oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw" id="space-context-btn-undefined" aria-expanded="false"><span class="oc-icon oc-icon-m oc-icon-passive"><!----></span></button> </div>
                  </div>
                </div>
                <!---->
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
  <side-bar-stub></side-bar-stub>
</div>
`;
